src/hooks/useFeatureToggling.ts

```diff
@@ -0,0 +1,126 @@
+import { useState } from 'react';
+
+import { useMsal } from '@azure/msal-react';
+import { useQuery } from '@tanstack/react-query';
+
+import { useAuth } from '../providers/AuthProvider/AuthProvider';
+import { auth, environment } from '../utils';
+
+const {
+  getAppName,
+  getEnvironmentName,
+  getApiUrl,
+  getPortalProdClientId,
+  getApiScope,
+} = environment;
+const { GRAPH_REQUESTS_BACKEND, acquireToken } = auth;
+
+// These three types (FeatureToggleDto, Feature, GraphUser) are from the swagger generated types in the portal API
+
+export type FeatureToggleDto = {
+  applicationName?: string | null;
+  features?: Array<Feature> | null;
+};
+
+export type Feature = {
+  uuid?: string | null;
+  featureKey?: string | null;
+  description?: string | null;
+  activeUsers?: Array<GraphUser> | null;
+  activeEnvironments?: Array<string> | null;
+};
+
+export type GraphUser = {
+  id?: string | null;
+  displayName?: string | null;
+  mail?: string | null;
+  userPrincipalName?: string | null;
+};
+
+interface useFeatureTogglingProps {
+  featureKey: string;
+}
+
+export function useFeatureToggling({ featureKey }: useFeatureTogglingProps) {
+  const [showContent, setShowContent] = useState<boolean | undefined>(
+    undefined
+  );
+  const { instance } = useMsal();
+  const account = useAuth().account;
```

Comment:
Just a suggeation but maybe this looks nicer? 😁
```suggestion
  const { account } = useAuth();
```